### PR TITLE
added supported sources section for all pages and updated introduction doc

### LINKED ISSUE #580: [Doc] Make 'supported data sources' information more accessible
## Documentation Scope
- [ ] All docs in this folder: https://devlake.apache.org/docs/Plugins
- [ ] Homepage: https://devlake.apache.org/

## Describe the Change
- Add a section called `Supported Versions` to each plugin's detail page 
image

- Update the wordings and link at the supported data sources section on the [homepage](https://devlake.apache.org/)
image

### PR DESCRIPTION
**Pull request by ghost**
### ⚠️ &nbsp;&nbsp;Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have `npm run build` and `npm run serve` locally before submitting this PR
- [x] I have read through the [Contributing](https://devlake.apache.org/community/) Documentation

# Summary

### Does this close any open issues?
Closes #580

### CODE REVIEW

File: docs/Plugins/bitbucket.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ This plugin collects various entities from Bitbucket, including pull requests, i
 
 As of v0.14.2, `bitbucket` plugin can only be invoked through DevLake API. Its support in Config-UI is WIP.
 
+## Supported Versions
+Available for BitBucket Cloud
```
Comment (Startrekzky) [inline docs/Plugins/bitbucket.md]: 1. Can you modify this line to 
```
"Available for BitBucket Cloud. Check [this doc](relative url of the supported data sources) for more details."
```

2. Add the URL to every plugin's supported versions.

On a side note, we can put the detailed versions here, but remember that we have to modify two docs once a new version has been supported for a plugin.

File: docs/Plugins/github.md [hunk 2]
```diff
@@ -12,6 +12,9 @@ This plugin collects GitHub data through [REST API](https://docs.github.com/en/r
 - How fast do we respond to customer requirements?
 - Was our quality improved or not?
 
+## Supported Versions
+Available for all versions
```
Comment (Startrekzky) [inline docs/Plugins/github.md]: Should it be GitHub Cloud? DevLake does not support GitHub Enterprise for now.

File: docs/Plugins/pagerduty.md [hunk 3]
```diff
@@ -14,6 +12,9 @@ This plugin collects all incidents from PagerDuty, and uses them to compute inci
 * [Incident Age](/Metrics/IncidentAge.md)
 * [Incident Count Per 1k Lines of Code](/Metrics/IncidentCountPer1kLinesOfCode.md)
 
+## Supported Versions
+Available for Cloud
```
Comment (Startrekzky) [inline docs/Plugins/pagerduty.md]: Available for PagerDuty Cloud. Check [this doc](relative url of the supported data sources) for more details.

File: docs/Plugins/sonarqube.md [hunk 4]
```diff
@@ -9,6 +9,9 @@ description: >
 This plugin collects SonarQube data through its REST APIs. SonarQube is a tool for static code analysis and code quality management. It can help you discover potential problems and defects in your code, and provide suggestions and solutions.
 (Please note that this version of SonarQube cannot collect issues with more than 10k+)
 
+## Supported Versions
+Available for v8.x, v9.x
```
Comment (Startrekzky) [inline docs/Plugins/sonarqube.md]: Available for SonarQube Server v8.x, v9.x. Check [this doc](relative url of the supported data sources) for more details.

File: docs/Plugins/zentao.md [hunk 5]
```diff
@@ -8,6 +8,9 @@ description: >
 
 This plugin collects Zentao data through its REST APIs. [Zentao](https://github.com/easysoft/zentaopms) is an issue-tracking tool similar to Jira.
 
+## Supported Versions
+Available for v17.x, v18.x
```
Comment (Startrekzky) [inline docs/Plugins/zentao.md]: Available for Zentao OSS v18.3, Enterprise v8.2, Flagship v4.3. Check [this doc](relative url of the supported data sources) for more details.

File: docs/Plugins/gitlab.md [hunk 6]
```diff
@@ -13,6 +13,9 @@ This plugin collects GitLab data through [API](https://docs.gitlab.com/ee/api/).
 - How long does it take for your codes to get merged?
 - How much time is spent on code review?
 
+## Supported Versions
+Available for GitLab Cloud, Community Edition 11+
```
Comment (Startrekzky) [inline docs/Plugins/gitlab.md]: Available for GitLab Cloud, Community Edition 11+. Check [this doc](relative url of the supported data sources) for more details.

File: docs/Plugins/jenkins.md [hunk 7]
```diff
@@ -12,6 +12,9 @@ This plugin collects Jenkins data through [Remote Access API](https://www.jenkin
 - What is the build success rate?
 - How long does it take for a code change to be deployed into production?
 
+## Supported Versions
+Versions 2.263.x+
```
Comment (Startrekzky) [inline docs/Plugins/jenkins.md]: Available for Jenkins v2.263.x+. Check [this doc](relative url of the supported data sources) for more details.

**Review by Startrekzky — CHANGES_REQUESTED** [review]
Please be more careful of the wording.

**Comment by Startrekzky** [comment]
Hi @klesh , some valid URLs didn't pass the link checker again. Can you take a look? No rush.

File: docs/Plugins/pagerduty.md (hunk 3 above)
Comment (ghost) [reply docs/Plugins/pagerduty.md]: Thanks @Startrekzky but I cannot find the relative url for PagerDuty. Let me know if we can create an issue.

Screenshot 2023-07-20 at 4 48 11 PM

File: docs/Plugins/github.md (hunk 2 above)
Comment (ghost) [reply docs/Plugins/github.md]: Sure, I have updated.

File: src/components/Sections/WhyDevlake.tsx [hunk 8]
```diff
@@ -102,10 +102,10 @@ function Supports() {
         className="text-label18 font-inter text-neutral-500 mt-6
         sm:text-label16 sm:mt-4"
       >
+        Check out all {" "}
         <C.InlineLink link="https://devlake.apache.org/docs/Overview/Roadmap">
```
Comment (Startrekzky) [inline src/components/Sections/WhyDevlake.tsx]: Please change the URL to the Supported data sources page

File: docs/Plugins/tapd.md [hunk 9]
```diff
@@ -8,6 +8,9 @@ description: >
 
 This plugin collects TAPD data through its REST APIs. TAPD is an issue-tracking tool similar to Jira.
 
+## Supported Versions
+Available for all versions
```
Comment (Startrekzky) [inline docs/Plugins/tapd.md]: Please add the "check this doc for more details" sentence to all the manuals. I noticed that some manuals are not added.

**Review by Startrekzky — CHANGES_REQUESTED** [review]

File: docs/Plugins/tapd.md (hunk 9 above)
Comment (ghost) [reply docs/Plugins/tapd.md]: Okay, sure.

**Comment by klesh** [comment]
Please merge https://github.com/apache/incubator-devlake-website/pull/600 first, and rebase this PR, it should be OK afterward.

**Comment by ghost** [comment]
@Startrekzky @klesh let me know if it worked.

**Comment by ghost** [comment]
@Startrekzky I've added introduction page updates to the same PR. Closes #593

File: src/components/Sections/WhyDevlake.tsx [hunk 10]
```diff
@@ -102,10 +102,10 @@ function Supports() {
         className="text-label18 font-inter text-neutral-500 mt-6
         sm:text-label16 sm:mt-4"
       >
-        <C.InlineLink link="https://devlake.apache.org/docs/Overview/Roadmap">
-          More data sources
-        </C.InlineLink>{" "}
-        coming soon...
+        Check out all {" "}
+        <C.InlineLink link="https://devlake.apache.org/docs/next/Overview/SupportedDataSources">
```
Comment (Startrekzky) [inline src/components/Sections/WhyDevlake.tsx]: It should point to the stable doc `https://devlake.apache.org/docs/Overview/SupportedDataSources`, not the next doc

**Review by Startrekzky — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@Startrekzky I've updated the changes.

**Review by Startrekzky — APPROVED** [review]
LGTM

**Pull request merged**